dids/src/main/kotlin/web5/sdk/dids/DidDht.kt

```diff
@@ -0,0 +1,424 @@
+package web5.sdk.dids
+
+import com.nimbusds.jose.JWSAlgorithm
+import com.nimbusds.jose.jwk.Curve
+import com.nimbusds.jose.jwk.JWK
+import com.nimbusds.jose.util.Base64URL
+import foundation.identity.did.DIDDocument
+import foundation.identity.did.Service
+import foundation.identity.did.VerificationMethod
+import io.ktor.client.engine.HttpClientEngine
+import org.xbill.DNS.DClass
+import org.xbill.DNS.Message
+import org.xbill.DNS.Name
+import org.xbill.DNS.Section
+import org.xbill.DNS.TXTRecord
+import web5.sdk.common.Convert
+import web5.sdk.common.ZBase32
+import web5.sdk.crypto.Crypto
+import web5.sdk.crypto.Ed25519
+import web5.sdk.crypto.KeyManager
+import web5.sdk.crypto.Secp256k1
+import java.net.URI
+
+/**
+ * Configuration for [DidDht].
+ *
+ * @property didDhtGateway The URL of the DID DHT gateway to use.
+ * @property engine The HTTP client engine to use for requests.
+ */
+public class DidDhtConfiguration internal constructor(
+  // TODO(gabe) update this with our own node when it's ready
+  public var didDhtGateway: String = "https://router.nuh.dev:6881",
+  public var engine: HttpClientEngine? = null,
+)
+
+/**
+ * Specifies options for creating a new "did:dht" Decentralized Identifier (DID).
+ * @property verificationMethodsToAdd A list of [JWK]s to add to the DID Document mapped to their purposes
+ * as verification methods.
+ * @property servicesToAdd A list of [Service]s to add to the DID Document.
+ * @property publish Whether to publish the DID Document to the DHT after creation.
+ */
+public class CreateDidDhtOptions(
+  public val verificationMethodsToAdd: Iterable<Pair<JWK, Array<PublicKeyPurpose>>>? = null,
```

Comment:
thoughts on `verificationMethods` instead of `verificationMethodsToAdd`? i feel like the `toAdd` is implied because it's `create`

Reply:
https://github.com/TBD54566975/web5-kt/pull/105#discussion_r1383759490